format/macho/macho.go

```diff
@@ -0,0 +1,868 @@
+		case LC_SYMTAB:
+			d.FieldU32("symoff")
+			d.FieldU32("nsyms")
+			d.FieldU32("stroff")
+			d.FieldU32("strsize")
+		case LC_DYSYMTAB:
+			d.FieldU32("ilocalsym")
+			d.FieldU32("nlocalsym")
+			d.FieldU32("iextdefsym")
+			d.FieldU32("nextdefsym")
+			d.FieldU32("iundefsym")
+			d.FieldU32("nundefsym")
+			d.FieldU32("tocoff")
+			d.FieldU32("ntoc")
+			d.FieldU32("modtaboff")
+			d.FieldU32("nmodtab")
+			d.FieldU32("extrefsymoff")
+			d.FieldU32("nextrefsyms")
+			d.FieldU32("indirectsymoff")
+			d.FieldU32("nindirectsyms")
+
+			d.FieldU32("extreloff")
+			d.FieldU32("nextrel")
+			d.FieldU32("locreloff")
+			d.FieldU32("nlocrel")
+		case LC_BUILD_VERSION:
+			d.FieldU32("platform")
+			d.FieldU32("minos")
+			d.FieldU32("sdk")
+			ntools := d.FieldU32("ntools")
+			var ntoolsIdx uint64
+			d.FieldStructArrayLoop("tools", "tool", func() bool {
+				return ntoolsIdx < ntools
+			}, func(d *decode.D) {
+				d.FieldU32("tool")
+				d.FieldU32("version")
+				ntoolsIdx++
+			})
+		case LC_CODE_SIGNATURE, LC_SEGMENT_SPLIT_INFO, LC_FUNCTION_STARTS, LC_DATA_IN_CODE, LC_DYLIB_CODE_SIGN_DRS, LC_LINKER_OPTIMIZATION_HINT:
+			d.FieldStruct("linkedit_data", func(d *decode.D) {
+				d.FieldU32("off")
+				d.FieldU32("size")
+			})
+		case LC_VERSION_MIN_IPHONEOS, LC_VERSION_MIN_MACOSX, LC_VERSION_MIN_TVOS, LC_VERSION_MIN_WATCHOS:
+			d.FieldU32("version")
+			d.FieldU32("sdk")
+		case LC_DYLD_INFO, LC_DYLD_INFO_ONLY:
+			d.FieldStruct("dyld_info", func(d *decode.D) {
+				d.FieldU32("rebase_off")
+				d.FieldU32("rebase_size")
+				d.FieldU32("bind_off")
+				d.FieldU32("bind_size")
+				d.FieldU32("weak_bind_off")
+				d.FieldU32("weak_bind_size")
+				d.FieldU32("lazy_bind_off")
+				d.FieldU32("lazy_bind_size")
+				d.FieldU32("export_off")
+				d.FieldU32("export_size")
+			})
+		case LC_MAIN:
+			d.FieldStruct("entrypoint", func(d *decode.D) {
+				d.FieldU64("entryoff")
+				d.FieldU64("stacksize")
+			})
+		case LC_SOURCE_VERSION:
+			d.FieldStruct("source_version_tag", func(d *decode.D) {
+				d.FieldU64("tag")
+			})
+		case LC_LINKER_OPTION:
+			d.FieldStruct("linker_option", func(d *decode.D) {
+				count := d.FieldU32("count")
+				d.FieldUTF8NullFixedLen("option", int(count))
+			})
+		case LC_ENCRYPTION_INFO, LC_ENCRYPTION_INFO_64:
+			d.FieldStruct("encryption_info", func(d *decode.D) {
+				d.FieldU32("offset")
+				d.FieldU32("size")
+				d.FieldU32("id")
+			})
+		case LC_IDFVMLIB, LC_LOADFVMLIB:
+			d.FieldStruct("fvmlib", func(d *decode.D) {
+				offset := d.FieldU32("offset")
+				d.FieldU32("minor_version")
+				d.FieldU32("header_addr", scalar.Hex)
+				d.FieldUTF8NullFixedLen("name", int(cmdsize)-int(offset))
+			})
+		default:
+			if _, ok := loadCommands[cmd]; !ok {
+				d.SeekRel(int64((cmdsize - 8) * 8))
+				// Seek Rel so the parts are marked unknown
+			}
+		}
+		ncmdsIdx++
+	})
+}
+
+func fatParse(d *decode.D) {
+	// Go to start of the file again
+	d.SeekAbs(0)
+	var narchs uint64
+	var ofile_offsets []uint64
```

Comment:
`ofileOffsets`? wonder if some other vars should be snakeCased also like `nArchs`?

Reply:
`ofileOffsets` fix is made sorry I was probably transitioning from work there and used snake_case 😆 
For `nfoobar` variable I would not go with camel case really

Reply:
😃  know the feeling, can be a bit confusing in this project, jq uses snake_cake